src/msftoolbox/reliefweb/reports.py

```diff
@@ -0,0 +1,253 @@
+        self,
+        app_name = "testing-rwapi",
+        preset = "latest",
+        limit = 50,
+        profile = "full"
+        ):
+        """Initialises the Relief Web Extractor
+
+        Args:
+            process_stage (str): The stage of the pipeline
+            source_type (str): The source of the data
+            destination_type (str): The destination of the data
+            datalake_url (str): The URL of the storage account.
+            datalake_container_name (str): The name of the container in the storage account.
+            app_name (str, optional): The name of the app. Required by Relief Web for their analytics. Defaults to "testing-rwapi".
+            preset (str, optional): A shorthand specification of sets of fields, filters and sort order for common use-cases.
+                                    Defaults to "latest", which sorts by date for appropriate content types. Countries and sources sorted by id. .
+            limit (int, optional): How many results to return. Defaults to 1000, max 1000.
+            profile (str, optional): A shorthand specification for which sets of fields to include in result.. Defaults to "full".
+        """
+        self.app_name = app_name
+        self.preset = preset
+        self.limit = limit
+        self.profile = profile
+
+    def try_key(
+        self,
+        key:str,
+        data:dict
+        ):
+        """Tests a key in a dictionary and resolves to None if missing
+
+        Args:
+            key (str): They key to test for
+            data (dict): The dictionary to test in
+
+        Returns:
+            any: The value associated with the key
+        """
+        if key in data.keys():
+            return data[key]
+        else:
+            print(f"Key {key} missing in article {data['title']}")
+            return None
+
+    def validate_date(
+        self,
+        date_string:str
+        ) -> bool:
+        """Returns a boolean indicating if the date string is in "%Y-%m-%d" format
+
+        Args:
+            date_string (str): The date as a string
+
+        Returns:
+            bool: True is a correctly formatted string and False is a wrongly formatted string
+        """
+        try:
+            # Try to create a datetime object from the date_str
+            # This checks both the format and the validity of the date (e.g., no February 30)
+            datetime.strptime(date_string, "%Y-%m-%d")
+            return True
+        except ValueError:
+            # If ValueError is raised, the format is wrong or the date is invalid
+            return False
+
+
+    def list_reports(
+        self,
+        start_date:str,
+        end_date:str,
+        query_value:str,
+        query_fields:list = ["body", "title"],
+        query_operator:str = "OR",
+        country_filter:list = None,
+        response_format:str = "dictionary"
+        ) -> list:
+        """List all reports matching a query with optional filters.
+
+        Args:
```

Comment:
Looking back at this, perhaps we should make the defaults explicit in the docstrings like we did in the init